src/nvim/path.c

```diff
@@ -2467,3 +2467,33 @@ void path_guess_exepath(const char *argv0, char *buf, size_t bufsize)
     xstrlcpy(buf, argv0, bufsize);
   }
 }
+
+/// Handles file path that starts with `file:[/|///]`
+///
+/// One and three slashes mean that file location is localhost/local.
+/// Example:
+/// - file:/path/to/file refers to /path/to/file
```

Comment:
is there any system besides KDE which has the single-slash variant? 

is there a [vscode test](https://github.com/microsoft/vscode-uri/blob/53e4ca6263f2e4ddc35f5360c62bc1b1d30f27dd/src/test/uri.test.ts#L264-L268) which asserts this? 

please add a reference, I guess https://en.wikipedia.org/wiki/File_URI_scheme#Number_of_slash_characters is fine.

Reply:
> is there any system besides KDE which has the single-slash variant?

Seems like this might be a good guide; http://eed3si9n.com/encoding-file-path-as-URI-reference/

Unix: u1/u3 `file:/` and `file:///`
Windows: u0/u3 `file:` and `file:///` (But, it seems like currently, neovim supports `

But, I don't know how to find if other system uses single-slash variant.  (googled a bit, but it leads to irrelevant stuff)

> is there a vscode test

This is trickier than I have thought before, as `:`(`0x3A`) and ` ` (`0x20`) are not allowed in file_path, Windows/DOS seems to convert those into hex code and Neovim needs to handle this case. 

For this one, it seems that `vim.uri_to_fname`([ref](https://github.com/neovim/neovim/blob/eadedfd9084c338a9162ef3706756bda443d6559/runtime/lua/vim/uri.lua#L102)) is available for file URI conversion, from lua side, but none seems to exist on C side? (I don't know if I'm understanding this correctly)

One thought is that it would be nice if c library side handles URIs nicely like lua side does. 


> please add a reference in this docstring, I guess https://en.wikipedia.org/wiki/File_URI_scheme#Number_of_slash_characters is fine.

Done.